src/SFML/Window/GlContext.cpp

```diff
@@ -1061,19 +1060,44 @@ void GlContext::checkSettings(const ContextSettings& requestedSettings) const
         (m_settings.depthBits < requestedSettings.depthBits) || (!m_settings.sRgbCapable && requestedSettings.sRgbCapable))
     {
-        err() << "Warning: The created OpenGL context does not fully meet the settings that were requested" << '\n'
-              << "Requested: version = " << requestedSettings.majorVersion << "." << requestedSettings.minorVersion
-              << " ; depth bits = " << requestedSettings.depthBits << " ; stencil bits = " << requestedSettings.stencilBits
-              << " ; AA level = " << requestedSettings.antialiasingLevel << std::boolalpha
-              << " ; core = " << ((requestedSettings.attributeFlags & ContextSettings::Core) != 0)
-              << " ; debug = " << ((requestedSettings.attributeFlags & ContextSettings::Debug) != 0)
-              << " ; sRGB = " << requestedSettings.sRgbCapable << std::noboolalpha << '\n'
-              << "Created: version = " << m_settings.majorVersion << "." << m_settings.minorVersion
-              << " ; depth bits = " << m_settings.depthBits << " ; stencil bits = " << m_settings.stencilBits
-              << " ; AA level = " << m_settings.antialiasingLevel << std::boolalpha
-              << " ; core = " << ((m_settings.attributeFlags & ContextSettings::Core) != 0)
-              << " ; debug = " << ((m_settings.attributeFlags & ContextSettings::Debug) != 0)
-              << " ; sRGB = " << m_settings.sRgbCapable << std::noboolalpha << std::endl;
+        log("Warning: The created OpenGL context does not fully meet the settings that were requested\n",
+            "Requested: version = ",
+            requestedSettings.majorVersion,
+            ".",
+            requestedSettings.minorVersion,
+            " ; depth bits = ",
+            requestedSettings.depthBits,
+            " ; stencil bits = ",
+            requestedSettings.stencilBits,
+            " ; AA level = ",
+            requestedSettings.antialiasingLevel,
+            std::boolalpha,
+            " ; core = ",
+            ((requestedSettings.attributeFlags & ContextSettings::Core) != 0),
+            " ; debug = ",
+            ((requestedSettings.attributeFlags & ContextSettings::Debug) != 0),
+            " ; sRGB = ",
+            requestedSettings.sRgbCapable,
+            std::noboolalpha,
+            '\n',
+            "Created: version = ",
+            m_settings.majorVersion,
+            ".",
+            m_settings.minorVersion,
+            " ; depth bits = ",
+            m_settings.depthBits,
+            " ; stencil bits = ",
+            m_settings.stencilBits,
+            " ; AA level = ",
+            m_settings.antialiasingLevel,
+            std::boolalpha,
+            " ; core = ",
+            ((m_settings.attributeFlags & ContextSettings::Core) != 0),
+            " ; debug = ",
+            ((m_settings.attributeFlags & ContextSettings::Debug) != 0),
+            " ; sRGB = ",
+            m_settings.sRgbCapable,
+            std::noboolalpha);
```

Comment:
This is one of the biggest downsides. clang-format is configured to put each function parameter on a separate line whereas chances of `operator<<` are more neatly packed into lines. This isn't a dealbreaker in my opinion but it is a bit of a bummer. If we really wanted to we could still make a local `std::stringstream` then pass that stream to `log()`.